src/main/kotlin/com/coder/jetbrains/services/CoderPortForwardService.kt

```diff
@@ -48,12 +56,44 @@ class CoderPortForwardService(
     }
 
     private fun start() {
-        logger.info("starting port scanner")
+        // TODO: make path configurable?
+        val devcontainerFile = File(System.getProperty("user.home"), ".cache/JetBrains/devcontainer.json")
+        if (devcontainerFile.exists()) {
+            try {
+                val json = devcontainerFile.readText()
+                val obj = JSONObject(json)
+
+                val portsAttributes = obj.optJSONObject("portsAttributes") ?: JSONObject()
+                portsAttributes.keys().forEach { spec ->
+                    portsAttributes.optJSONObject(spec)?.let { attrs ->
+                        val onAutoForward = attrs.optString("onAutoForward")
```

Comment:
Nit/Not even sure I want a change here: spec says "onAutoForward" is an an enum of `[
										"notify",
										"openBrowser",
										"openBrowserOnce",
										"openPreview",
										"silent",
										"ignore"
									]`. Might be worth checking we don't set a `PortMatcher('[object Object]')` or something to save us a debugging headache in the future, but also see the value of leaving it flexible so we don't have to come back and make an update in the event the spec changes

Reply:
I added validation that the `onAutoForward` value is a string. I think it probably makes sense to accept any non-empty value here except `ignore` as equivalent to `notify`, but happy to tweak this as you see fit.